pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -210,27 +226,33 @@ func (gb *GraphBuilder) syncMonitors(resources map[schema.GroupVersionResource]s
 			errs = append(errs, fmt.Errorf("couldn't look up resource %q: %v", resource, err))
 			continue
 		}
-		c, s, err := gb.controllerFor(resource, kind)
+		shared, err := gb.informerFor(resource, kind)
 		if err != nil {
 			errs = append(errs, fmt.Errorf("couldn't start monitor for resource %q: %v", resource, err))
 			continue
 		}
-		current[resource] = &monitor{store: s, controller: c}
+		current[resource] = shared
 		added++
 	}
 	gb.monitors = current
 
-	for _, monitor := range toRemove {
```

Comment:
This seems like it would have been an ideal place to explicitly stop the started informer based on no longer wanting to watch this resource. Is there a reason we can't be more direct with stopping informers for resources no longer listed in discovery?

Reply:
fixed, see above explanation

I think doing it in `startMonitors` is sufficient and prevents an unnecessary indirection